### PR TITLE
Fix 10922: array type in foreach

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/phpstan/phpstan/issues/10922

### CODE REVIEW

File: src/Analyser/NodeScopeResolver.php [hunk 1]
```diff
@@ -951,7 +951,10 @@ private function processStmtNode(
 						]),
 					)));
 				} else {
```
Comment (VincentLanglet) [inline src/Analyser/NodeScopeResolver.php]: I'm not sure about the purpose of 
```
if ($this->polluteScopeWithAlwaysIterableForeach) {
     $finalScope = $finalScope->mergeWith($scope->filterByTruthyValue(new BooleanOr(
	     new BinaryOp\Identical(
		     $stmt->expr,
			 new Array_([]),
	     ),
		 new FuncCall(new Name\FullyQualified('is_object'), [
			 new Arg($stmt->expr),
		 ]),
	)));
} else {
	$finalScope = $finalScope->mergeWith($scope);
}
```

But I soon as I make this change, the issue is solved.

I dunno
- if it has a bad impact (I can add tests if you had some in mind)
- if I should just remove the else part and the if condition.

File: src/Analyser/NodeScopeResolver.php [hunk 2]
```diff
@@ -951,7 +951,10 @@ private function processStmtNode(
...
-					$finalScope = $finalScope->mergeWith($scope);
+					$finalScope = $finalScope->mergeWith($scope->filterByTruthyValue(new BinaryOp\Identical(
+						$stmt->expr,
+						new Array_([]),
+					)));
```
Comment (ondrejmirtes) [inline src/Analyser/NodeScopeResolver.php]: Please remove the if-else as both these branches look like should be the same.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Analyser/Bug10922Test.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		...$args,
+	): void
+	{
+		$this->assertFileAsserts($assertType, $file, ...$args);
+	}
+
+	public static function getAdditionalConfigFiles(): array
+	{
+		return [
+			__DIR__ . '/../../../conf/bleedingEdge.neon',
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/Bug10922Test.php:31]: Merging with parent would be better (I think bleedingEdge is there already)

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Analyser/Bug10922Test.php (hunk 3 above)
Comment (VincentLanglet) [reply tests/PHPStan/Analyser/Bug10922Test.php:31]: Parent is
```
/**
	 * @return string[]
	 */
	public static function getAdditionalConfigFiles(): array
	{
		return [];
	}
```

Comment (VincentLanglet) [reply tests/PHPStan/Analyser/Bug10922Test.php:31]: NodeScopeResolverTest use
```
	public static function getAdditionalConfigFiles(): array
	{
		return [
			__DIR__ . '/../../../conf/bleedingEdge.neon',
			__DIR__ . '/typeAliases.neon',
		];
	}
```
I followed the same idea

File: src/Analyser/NodeScopeResolver.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Analyser/NodeScopeResolver.php]: done

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**